src/python/pants/backend/python/subsystems/pex_build_util.py

```diff
@@ -92,39 +95,29 @@ def identify_missing_init_files(sources):
   return {os.path.join(package, '__init__.py') for package in packages} - set(sources)
 
 
-def _create_source_dumper(builder, tgt):
-  if type(tgt) == Files:
-    # Loose `Files` as opposed to `Resources` or `PythonTarget`s have no (implied) package structure
-    # and so we chroot them relative to the build root so that they can be accessed via the normal
-    # python filesystem APIs just as they would be accessed outside the chrooted environment.
-    # NB: This requires we mark the pex as not zip safe so these `Files` can still be accessed in
-    # the context of a built pex distribution.
-    chroot_path = lambda relpath: relpath
-    builder.info.zip_safe = False
-  else:
-    chroot_path = lambda relpath: os.path.relpath(relpath, tgt.target_base)
-
-  dump = builder.add_resource if has_resources(tgt) else builder.add_source
+def _create_source_dumper(builder: PEXBuilder, tgt: Target) -> Callable[[str], None]:
   buildroot = get_buildroot()
-  return lambda relpath: dump(os.path.join(buildroot, relpath), chroot_path(relpath))
 
+  def get_chroot_path(relpath: str) -> str:
+    if type(tgt) == Files:
+      # Loose `Files`, as opposed to `Resources` or `PythonTarget`s, have no (implied) package
+      # structure and so we chroot them relative to the build root so that they can be accessed
+      # via the normal Python filesystem APIs just as they would be accessed outside the
+      # chrooted environment. NB: This requires we mark the pex as not zip safe so
+      # these `Files` can still be accessed in the context of a built pex distribution.
+      builder.info.zip_safe = False
+      return relpath
+    return str(Path(relpath).relative_to(tgt.target_base))
 
-def dump_sources(builder, tgt, log):
```

Comment:
I deleted this because there were no usages and it's not a public API. It's highly similar to `add_sources_from()`, which we do use a lot of.